pkg/lib/bundle/generate.go

```diff
@@ -63,37 +82,83 @@ func GenerateFunc(directory, packageName, channels, channelDefault string, overw
 		return err
 	}
 
-	file, err := ioutil.ReadFile(filepath.Join(directory, MetadataDir, AnnotationsFile))
-	if os.IsNotExist(err) || overwrite {
-		err = WriteFile(AnnotationsFile, filepath.Join(directory, MetadataDir), content)
-		if err != nil {
-			return err
-		}
-	} else if err != nil {
+	// Push the output yaml content to the correct directory and conditionally copy the manifest dir
+	outManifestDir, outMetadataDir, err := CopyYamlOutput(content, directory, outputDir, workingDir, overwrite)
+	if err != nil {
 		return err
-	} else {
-		log.Info("An annotations.yaml already exists in directory")
-		if err = ValidateAnnotations(file, content); err != nil {
-			return err
-		}
 	}
 
 	log.Info("Building Dockerfile")
 
 	// Generate Dockerfile
-	content, err = GenerateDockerfile(mediaType, ManifestsDir, MetadataDir, packageName, channels, channelDefault)
+	content, err = GenerateDockerfile(mediaType, ManifestsDir, MetadataDir, outManifestDir, outMetadataDir, workingDir, packageName, channels, channelDefault)
 	if err != nil {
 		return err
 	}
 
-	err = WriteFile(DockerFile, directory, content)
-	if err != nil {
+	_, err = os.Stat(filepath.Join(workingDir, DockerFile))
+	if os.IsNotExist(err) || overwrite {
+		err = WriteFile(DockerFile, workingDir, content)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
 		return err
+	} else {
+		log.Info("A bundle.Dockerfile already exists in current working directory")
 	}
 
 	return nil
 }
 
+// CopyYamlOutput takes the generated annotations yaml and writes it to disk.
+// If an outputDir is specified, it will copy the input manifests
+// It returns two strings. resultMetadata is the path to the output metadata/ folder.
+// resultManifests is the path to the output manifests/ folder -- if no copy occured,
+// it just returns the input manifestDir
+func CopyYamlOutput(annotationsContent []byte, manifestDir, outputDir, workingDir string, overwrite bool) (resultManifests, resultMetadata string, err error) {
+	// First, determine the parent directory of the metadata and manifest directories
+	copyDir := ""
+
+	// If an output directory is not defined defined, generate metadata folder into the same parent dir as existing manifest dir
+	if outputDir == "" {
+		copyDir = filepath.Dir(manifestDir)
+		resultManifests = manifestDir
+	} else { // otherwise copy the manifests into $outputDir/manifests and create the annotations file in $outputDir/metadata
+		copyDir = outputDir
+
+		log.Info("Generating output manifests directory")
+
+		resultManifests = filepath.Join(copyDir, "/manifests/")
+		// copy the manifest directory into $pwd/manifests/
+		err := copyManifestDir(manifestDir, resultManifests, overwrite)
+		if err != nil {
+			return "", "", err
+		}
+	}
+
+	// Now, generate the `metadata/` dir and write the annotations
+	file, err := ioutil.ReadFile(filepath.Join(copyDir, MetadataDir, AnnotationsFile))
```

Comment:
Instead of making the metadata and manifests directory match up, have we thought about inlining the annotations file and hiding it from the end-user completely?

Reply:
This is a valid point that Eric/SDK suggested a while back about inline annotations/Dockerfile during docker build command instead writing them in disk. However, it is still a valid use case to write those files to disk if someone wants to see them.

Reply:
You mean:
```
/manifests
annotations.json
```
?

Reply:
I still think we want this thing to be "Generate/scaffold this once, then you can do what you want with it". Today it's true that everything in `metadata/` is probably not super important to end users, but if we want to move lots of other metadata in the future that users DO want to configure, I think it's important to make it visible now.